File: 204.单例/Solution.java
class Solution {
    /**
     * @return: The same instance of this class every time
     */
    private static class Singleton{
        private static final Solution INSTANCE = new Solution();
    }
    private Solution () {}
    public static final Solution getInstance() {
        // write your code here
        return Singleton.INSTANCE;
    }
};

File: 204.单例/Solution.cpp
class Solution {
public:
    /**
     * @return: The same instance of this class every time
     */
    static Solution* getInstance() {
        // write your code here
        static Solution INSTANCE;
        return &INSTANCE;
    }
    
protected:
    struct Object_Creator{
        Object_Creator(){Solution::getInstance();}
    };
    static Object_Creator _object_creator;
    
    Solution() {}
    ~Solution() {}
};